text/0089-klt-context-information.md

```diff
@@ -0,0 +1,45 @@
+# KEP 89: Context Information in Keptn Tasks 
+**State: DRAFTING**
+
+## Dependencies
+* This KEP depends on KEP88 which introduces instances of Keptn Workloads and Apps
+
+## Motivation
+Currently, the lifecycle toolkit passes some context information (Workload, Application, Version) to the tasks. However, this information is not sufficient for promoting instances. As there should be a Keptn Task for promoting an instance, the context information should be extended to include the instance and a bit more information.
+
+This KEP proposes the extension of passed context information.
+
+## Current Design
+At the moment, the context information is passed as a JSON string in the `KEPTN_CONTEXT` environment variable. The JSON string contains the following information:
+```golang
+  type TaskContext struct {
+    WorkloadName    string `json:"workloadName"`
+    AppName         string `json:"appName"`
+    AppVersion      string `json:"appVersion"`
+    WorkloadVersion string `json:"workloadVersion"`
+    TaskType        string `json:"taskType"`
+    ObjectType      string `json:"objectType"`
+```
+
+This information is passed to the tasks as a JSON string in the `KEPTN_CONTEXT` environment variable which can be used in the corresponding tasks.
+
+## Assumptions / Definitions
+* It might be easier to pass over the whole KeptnAppVersion or KeptnWorkloadInstance as a JSON string instead of passing over the individual fields.
+* There is no sensitive information in the context information
```

Comment:
I would follow [RFC2119](https://www.ietf.org/rfc/rfc2119.txt) and have extra emphasis on this requirement
```suggestion
* There MUST NOT be sensitive information in the context information
```